cmd/entrypoints/serve.go

```diff
@@ -216,7 +198,20 @@ func serveGatewayInsecure(ctx context.Context, cfg *config.ServerConfig) error {
 	if err != nil {
 		return err
 	}
-	err = http.ListenAndServe(cfg.GetHostAddress(), httpServer)
+
+	var handler http.Handler
+	if cfg.Security.AllowCors {
+		handler = handlers.CORS(
+			handlers.AllowCredentials(),
+			handlers.AllowedHeaders(cfg.Security.AllowedHeaders),
```

Comment:
how does this work if admin is misconfigured and either the allowed headers or origins happens to be empty?

Reply:
I tested this. It will still work (I can cURL from Postman). But it will make it not possible for the UI to make valid CORS-enabled calls.
The domain needs to be user-configured no matter what. But I'm considering changing it so that the content-type header is always sent. We need it for the UI in all cases.